Detected package version outside of dependency constraint

I use Maui v10, when I compile I get the following warnings:

v3.0.2 (my new situation):

```log
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.12.2.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.SavedState.SavedState.Ktx 1.3.1.1 requires Xamarin.AndroidX.SavedState (>= 1.3.1.1 && < 1.3.2) but version Xamarin.AndroidX.SavedState 1.4.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime.Android (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime.Android 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Process 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.12.2.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.SavedState.SavedState.Ktx 1.3.1.1 requires Xamarin.AndroidX.SavedState (>= 1.3.1.1 && < 1.3.2) but version Xamarin.AndroidX.SavedState 1.4.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime.Android (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime.Android 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Process 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.12.2.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.SavedState.SavedState.Ktx 1.3.1.1 requires Xamarin.AndroidX.SavedState (>= 1.3.1.1 && < 1.3.2) but version Xamarin.AndroidX.SavedState 1.4.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime.Android (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime.Android 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
2>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Process 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.Runtime 2.10.0.1 was resolved.
```

v2.3.1 (my old situation):

```log
1>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Service 2.8.5.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.8.5.1 && < 2.8.6) but version Xamarin.AndroidX.Lifecycle.Runtime 2.9.2.1 was resolved.
1>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Service 2.8.5.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.8.5.1 && < 2.8.6) but version Xamarin.AndroidX.Lifecycle.Runtime 2.9.2.1 was resolved.
1>.\Some.csproj : warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.Service 2.8.5.1 requires Xamarin.AndroidX.Lifecycle.Runtime (>= 2.8.5.1 && < 2.8.6) but version Xamarin.AndroidX.Lifecycle.Runtime 2.9.2.1 was resolved.
```

I aim to keep warnings to a minimum or non-existing. So ideally, this package should be free of compiler warnings. Maybe co-pilot can solve this for you.

Trying to update all of my Nuget packages.  Everything successful except for Plugin.Admob going from 2.3.2-beta.7   to  Latest Stable 3.0.2.


warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.1 was resolved.
warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Activity.Ktx 1.10.1.3 requires Xamarin.AndroidX.Activity (>= 1.10.1.3 && < 1.10.2) but version Xamarin.AndroidX.Activity 1.12.2.1 was resolved.
warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Lifecycle.ViewModel.Ktx 2.9.2.1 requires Xamarin.AndroidX.Lifecycle.ViewModel (>= 2.9.2.1 && < 2.9.3) but version Xamarin.AndroidX.Lifecycle.ViewModel 2.10.0.1 was resolved.
(and several more)

And also this.....

1>MSBUILD : java.exe error JAVA0000: Error in C:\Users\...\.nuget\packages\xamarin.androidx.compose.runtime.annotation.android\1.10.0.1\buildTransitive\net10.0-android36.0\..\..\aar\runtime-annotation-android.aar:classes.jar:androidx/compose/runtime/Immutable.class:
1>MSBUILD : java.exe error JAVA0000: **Type androidx.compose.runtime.Immutable is defined multiple times:** C:\Users\mrray\.nuget\packages\xamarin.androidx.compose.runtime.annotation.android\1.10.0.1\buildTransitive\net10.0-android36.0\..\..\aar\runtime-annotation-android.aar:classes.jar:androidx/compose/runtime/Immutable.class, C:\Users\...\.nuget\packages\xamarin.androidx.compose.runtime.annotation.jvm\1.10.0.1\buildTransitive\net10.0-android36.0\..\..\jar\runtime-annotation-jvm.jar:androidx/compose/runtime/Immutable.class
1>MSBUILD : java.exe error JAVA0000: Compilation failed
1>MSBUILD : java.exe error JAVA0000: java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: C:\Users\...\.nuget\packages\xamarin.androidx.compose.runtime.annotation.android\1.10.0.1\buildTransitive\net10.0-android36.0\..\..\aar\runtime-annotation-android.aar:classes.jar
1>MSBUILD : java.exe error JAVA0000: androidx/compose/runtime/Immutable.class
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.internal.xv.a(R8_8.11.18_94248c9a6fd57686ded562f231d18f14e8923ad1fe977e99b974a9bf901c8a48:131)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.main(R8_8.11.18_94248c9a6fd57686ded562f231d18f14e8923ad1fe977e99b974a9bf901c8a48:5)
1>MSBUILD : java.exe error JAVA0000: Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: C:\Users\...\.nuget\packages\xamarin.androidx.compose.runtime.annotation.android\1.10.0.1\buildTransitive\net10.0-android36.0\..\..\aar\runtime-annotation-android.aar:classes.jar:androidx/compose/runtime/Immutable.class
1>MSBUILD : java.exe error JAVA0000: 	at Version.fakeStackEntry(Version_8.11.18.java:0)

--- a/docs/Troubleshooting-Android.md
+++ b/docs/Troubleshooting-Android.md
@@ -58,6 +58,43 @@ Make sure your app explicitly sets a minimum Android platform version compatible
 
 - [`#68`](https://github.com/marius-bughiu/Plugin.AdMob/issues/68) (namespace warnings when min platform is too low)
 
+## NU1608 warnings: "Detected package version outside of dependency constraint"
+
+### Symptoms
+
+Building an app that references the plugin emits warnings like:
+
+```
+warning NU1608: Detected package version outside of dependency constraint: Xamarin.AndroidX.Fragment.Ktx 1.8.8.1 requires Xamarin.AndroidX.Fragment (>= 1.8.8.1 && < 1.8.9) but version Xamarin.AndroidX.Fragment 1.8.9.1 was resolved.
+```
+
+### Root cause
+
+Google Play Services (pulled in via `Xamarin.GooglePlayServices.Ads.Lite`) depends on older `Xamarin.AndroidX.*.Ktx` packages, whose declared upper bounds don't admit the newer AndroidX base packages that .NET MAUI resolves. The pairs are compatible in practice — NU1608 is a warning, not an error — but the noise is legitimate to want gone.
+
+### Fix / mitigation
+
+Reference the `Ktx` packages directly in your app project at versions aligned with the resolved base packages. Verified to produce zero NU1608 warnings with Plugin.AdMob 10.0.80-beta.7 and .NET MAUI 10.0.80 (the sample apps carry the same set):
+
+```
+<ItemGroup Condition="'$(TargetFramework)' == 'net10.0-android'">
+	<PackageReference Include="Xamarin.AndroidX.Fragment.Ktx" Version="1.8.9.4" />
+	<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.6.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.13.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel.Ktx" Version="2.11.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.SavedState.SavedState.Ktx" Version="1.5.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android" Version="2.11.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx" Version="2.11.0.1" />
+	<PackageReference Include="Xamarin.AndroidX.Lifecycle.Process" Version="2.11.0.1" />
+</ItemGroup>
+```
+
+If new NU1608 warnings appear after a MAUI or plugin update, adjust the pinned versions to match the newly resolved base packages (check the warnings themselves or `obj/project.assets.json`).
+
+### Related issues
+
+- [`#82`](https://github.com/marius-bughiu/Plugin.AdMob/issues/82) (NU1608 dependency constraint warnings)
+
 ## Full-screen test ads fail with "no fill" on emulators (banners work)
 
 ### Symptoms

--- a/samples/Foo.Bar.SampleApp.Hybrid/Foo.Bar.SampleApp.Hybrid.csproj
+++ b/samples/Foo.Bar.SampleApp.Hybrid/Foo.Bar.SampleApp.Hybrid.csproj
@@ -57,6 +57,18 @@
         <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="10.0.9" />
     </ItemGroup>
 
+    <!-- Aligns the AndroidX Ktx packages with the resolved base package versions to silence NU1608 warnings. See: https://github.com/marius-bughiu/Plugin.AdMob/issues/82 -->
+    <ItemGroup Condition="'$(TargetFramework)' == 'net10.0-android'">
+        <PackageReference Include="Xamarin.AndroidX.Fragment.Ktx" Version="1.8.9.4" />
+        <PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.6.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.13.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel.Ktx" Version="2.11.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.SavedState.SavedState.Ktx" Version="1.5.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android" Version="2.11.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx" Version="2.11.0.1" />
+        <PackageReference Include="Xamarin.AndroidX.Lifecycle.Process" Version="2.11.0.1" />
+    </ItemGroup>
+
 	<ItemGroup>
 		<ProjectReference Include="..\..\src\Plugin.AdMob\Plugin.AdMob.csproj" />
 	</ItemGroup>

--- a/samples/Foo.Bar.SampleApp/Foo.Bar.SampleApp.csproj
+++ b/samples/Foo.Bar.SampleApp/Foo.Bar.SampleApp.csproj
@@ -55,6 +55,18 @@
 		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="10.0.9" />
 	</ItemGroup>
 
+	<!-- Aligns the AndroidX Ktx packages with the resolved base package versions to silence NU1608 warnings. See: https://github.com/marius-bughiu/Plugin.AdMob/issues/82 -->
+	<ItemGroup Condition="'$(TargetFramework)' == 'net10.0-android'">
+		<PackageReference Include="Xamarin.AndroidX.Fragment.Ktx" Version="1.8.9.4" />
+		<PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.6.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.13.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.Lifecycle.ViewModel.Ktx" Version="2.11.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.SavedState.SavedState.Ktx" Version="1.5.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx.Android" Version="2.11.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime.Ktx" Version="2.11.0.1" />
+		<PackageReference Include="Xamarin.AndroidX.Lifecycle.Process" Version="2.11.0.1" />
+	</ItemGroup>
+
 	<ItemGroup>
 	  <ProjectReference Include="..\..\src\Plugin.AdMob\Plugin.AdMob.csproj" />
 	</ItemGroup>

